docs/proposals/20201125-jaeger-traces-store.md

```diff
@@ -0,0 +1,124 @@
+# Jaeger traces store
+
+* **Owners:**:
+    * `@pavolloffay`
+
+* **Related Tickets:**
+    * [observatorium/issues/442](https://github.com/observatorium/observatorium/issues/442)
+    * Depends on: [Traces ingestion API and OpenTelemetry collector](https://github.com/observatorium/observatorium/pull/443)
+
+* **Other docs:**
+
+## Table of Contents
+
+- [TLDR](#tldr)
+- [Why](#why)
+- [Goals](#goals)
+- [Non-Goals](#non-goals)
+- [How](#how)
+    * [Jaeger configuration](#jaeger-configuration)
+        + [Single Jaeger instance](#single-jaeger-instance)
+            - [OpenTelemetry collector configuration](#opentelemetry-collector-configuration)
+        + [Jaeger instance per tenant](#jaeger-instance-per-tenant)
+            - [OpenTelemetry collector configuration](#opentelemetry-collector-configuration)
+    * [Jaeger Query](#jaeger-query)
+        + [Expose Jaeger Query API](#expose-jaeger-query-api)
+        + [Expose Jaeger Query UI](#expose-jaeger-ui)
+- [Action Plan](#action-plan)
+- [References](#references)
+
+<small>
+
+<i>
+
+<a href="http://ecotrust-canada.github.io/markdown-toc/">
+Table of contents generated with markdown-toc
+</a>
+
+</i>
+
+</small>
+
+## TLDR
+
+This proposal adds Jaeger deployment with a persistent storage (e.g. Elasticsearch) to store trace data and it as well defines a query API to search for traces.
+
+## Why
+
+Traces store adds a crucial capability to Observatorium to persist and query tracing data.
+
+## Goals
+
+* Configure OpenTelemetry collector to export data to Jaeger collector.
+* Deploy Jaeger with a persistent storage.
+* Define query API for trace data.
+
+## Non-Goals
+
+* Define trace ingestion api - it is done in [Traces ingestion API and OpenTelemetry collector](https://github.com/observatorium/observatorium/pull/443) proposal.
+
+## How
+
+The how is split into the following sections:
+* Jaeger configuration - describes how Jaeger components are deployed.
+* Jaeger Query - describes how users access data stored in Jaeger.
+
+### Jaeger configuration
+
+Currently, the most adopted and supported storage in Jaeger is Elasticsearch. Red Hat also supports Jaeger with Elasticsearch (6.x) as part of the OpenShift product. Therefore, this proposal assumes Elasticsearch will be used as a storage backend. Distributed tracing team at Red Hat is looking at alternative storages hence the storage technology can change in the future.
```

Comment:
Since this should be primarily community-driven effort, maybe mentioning Red Hat is not relevant. Could we rephrase this in a way that would say that Elasticsearch is the most used storage solution community-wide? (If we have any 'evidence' to support that statement that would be a cherry on top).

Reply:
> this in a way that would say that Elasticsearch is the most used storage solution community-wide? 

The text already mentions this. I have added a Red Hat there bc the majority of deployed Observatorium instances are seems to be at Red Hat. Red Hat is also mentioned at various parts in the existing docs.

Reply:
Red Hat also don't support Elasticsearch at this point, so not sure what's the point of those statements. 🤔

Reply:
While it's true we might be the biggest known user, I believe we still should have proposals which are formulated neutrally and take into account needs to the community as a whole. I'm not sure if we justify choices in other proposals by saying Red Hat uses a particular technology - I believe not, but I might be wrong. For me, it would be a stronger statement on why we decided to support a particular storage solution.

Reply:
ES support

https://docs.openshift.com/container-platform/4.9/logging/cluster-logging-release-notes.html
https://access.redhat.com/support/policy/updates/openshift#logging
https://catalog.redhat.com/software/containers/openshift-logging/elasticsearch-rhel8-operator/5fd118c5af03b18dd4ff3fc1?container-tabs=gti